### ISSUE TITLE
Job Instance and Job History API timestamps inconsistent with other timestamps

### ISSUE THREAD

**Issue by ketan** [gocd/gocd#10713]
##### Issue Type

- Bug Report

##### Summary

The current job instance API renders timestamps as milliseconds since unix epoch. This is inconsistent with other APIs that return an ISO8601 timestamp in the usual format (`YYYY-MM-DD'T'HH:mm:ssZ`) format.

The [agent job run history API](https://api.gocd.org/current/#agent-job-run-history) seems to render `job_state_transitions` in a slightly different format from the [get job instance API](https://api.gocd.org/current/#get-job-instance) and the [get job history api](https://api.gocd.org/current/#get-job-history).

Now the open question is if we should bump the API version for these APIs to break compatibility with previous version?

**Comment by ketan** [comment]
Seems to be the same case with:

- https://api.gocd.org/current/#get-pipeline-instance
- https://api.gocd.org/current/#get-pipeline-history
- https://api.gocd.org/current/#get-stage-instance
- https://api.gocd.org/current/#get-stage-history

**Comment by chadlwilson** [comment]
Yeah, I noted that in https://github.com/gocd/gocd/pull/10576#issuecomment-1193149267 but thanks for raising it separately. Not sure on historical context or approach to these things but I sense generally new API versions were introduced for incompatible changes like this would be?
